File: desh/ege2026kp/25solve/25-246.py
def isPrime( n ):
  return n > 1 and all( n % d for d in range(2, round(n**0.5)+1) )

def allDivs( n ):
  divs = set()
  for d in range(1, round(n**0.5)+1):
    if n % d == 0:
       divs.add( d )
       divs.add( n//d )
  return sorted(divs)

count = 0
n = 100000000 + 1
while count < 5:
  divs = allDivs( n )
  primeDivs = [d for d in divs if isPrime(d)]
  evenDivs = [d for d in divs if d % 2 == 0]
  if len(primeDivs) == len(evenDivs):
    print( n, abs( sum(primeDivs) - sum(evenDivs) ) )
    count += 1
  n += 1

File: desh/ege2026kp/25solve/25-246.pas
// Автор: И.В.Свистун
##
uses school;
var ch:=0; var n:=100000001;
while (ch<5) do
Begin    
    if (n.divisors.count(t ->t.IsEven)=n.divisors.count(t ->t.IsPrime))  then
  begin
    var pn:=n.divisors.where(t ->t.IsEven).sum;
    var en:=n.divisors.where(t ->t.IsPrime).sum;
    println(n, abs(pn-en));  ch+=1;
  end; 
  n+=1;
end